libminifi/include/io/Serializable.h

```diff
@@ -44,8 +44,9 @@ Integral byteSwap(Integral i) {
 #ifdef htonll
   return htonll(i);
 #else
-  #define htonll_r(x) ((((uint64_t)htonl(x)) << 32) + htonl((x) >> 32))
-  return htonll_r(i);
+  uint32_t lower_32_bits = static_cast<uint32_t>(i);
```

Comment:
are these two implementations equivalent? on big-endian systems `htonll` is noop while the other one swaps the bytes anyway

Reply:
similarly, the other `byteSwap` overloads all seem to be noop on big-endian systems

Reply:
The new code is equivalent to the old code which used the `#define`, but you are right that it's not always equivalent to `htonll`.  We check `is_little_endian` every time before we call `byteSwap`; it might be better to move that check inside `byteSwap`, but I didn't think that was in scope for this change.

Reply:
agree that it is not in scope
👍